inferred-spans/src/main/java/io/opentelemetry/contrib/inferredspans/internal/SamplingProfiler.java

```diff
@@ -723,7 +725,19 @@ public void copyFromFiles(Path activationEvents, Path traces) throws IOException
 
   @SuppressWarnings("FutureReturnValueIgnored")
   public void start() {
-    scheduler.submit(this);
+    profilingTask = scheduler.submit(this);
+  }
+
+  @SuppressWarnings({"FutureReturnValueIgnored", "Interruption"})
+  public void reschedule() {
+    Future<?> future = this.profilingTask;
+    if (future != null) {
+      if (future.cancel(true)) {
```

Comment:
```suggestion
      if (future.isDone() || future.cancel(true)) {
```

So the javadoc for `Future.cancel()` says that it returns 

> "`false` if the task could not be cancelled, typically because it has already completed;

If the thing finishes right before/when this `reschedule()` is invoked, we still want to kick it off again, right?

Reply:
When the task has already completed, it has already scheduled the next task, so on isDone() we do NOT want to schedule a new task in this reschedule() method